subworkflows/local/tool_specific_checks/main.nf

```diff
@@ -0,0 +1,30 @@
+workflow TOOL_SPECIFIC_CHECKS {
+    main:
+    def checkPathParamList = []
+
+    // Conditionally add tools specific parameters (only examples for now)
+    if (params.tools?.split(',')?.contains('snpeff') || params.tools?.split(',')?.contains('merge')) {
+        checkPathParamList.add(params.snpeff_cache)
+    }
+    if (params.tools?.split(',')?.contains('vep') || params.tools?.split(',')?.contains('merge')) {
+        checkPathParamList.add(params.vep_cache)
+    }
+
+    if (checkPathParamList) {
+        paths_to_check_channel = Channel.fromList(checkPathParamList)
+
+        missing_paths = paths_to_check_channel.map { path_item ->
+            def file_path = path_item instanceof Path ? path_item : file(path_item)
+            return [file_path, file_path.exists()]
+        }
+        .filter { !it[1] }
+        .collect()
+        .subscribe { missing ->
+            if (missing.size() > 0) {
+               def error_string = "The following required paths do not exist:\n" +
+                                   missing.collect { "  - ${it[0]}" }.join('\n')
+               error(error_string)
+            }
+        }
```

Comment:
@nvnieuwk is there a way to do this with the schema at the moment? If params.tools == vep, then vep_cache is required? 
Would be nicer to get the failures super early and not having to wait for the head job submission

Reply:
It is possible in the JSON schema spec (see [`if-then-else`](https://json-schema.org/understanding-json-schema/reference/conditionals#ifthenelse)), but this is not supported by Nextflow and nf-core tools (except for nf-schema). This will also not be part of the new schema specification for the parameters we are drafting. So it's not recommended to use this feature for now. We might add it some time in the future if we get all tooling to be compatible but that's low priority right now